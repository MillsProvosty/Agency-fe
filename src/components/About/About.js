import React from "react";
import "./About.scss";
import { NavLink } from "react-router-dom";
import Nav from "../Nav/Nav";

const About = () => {
  return (
    <section className="about">
      <Nav />
      <p className="first">
        <span>Agency</span> is a support networking service.
      </p>
      <p className="second">To begin..</p>
      <section className="options">
        <section className="option-1">
          <p>
            select the <span>Volunteer</span> button
          </p>
          <p>if you can lend a hand...</p>
          <NavLink to="/user-form">
            <button className="about-button">Volunteer</button>
          </NavLink>
        </section>
        <section className="option-2">
          <p>
            or, select the <span>Client</span> button
          </p>
          <p>if you would like some help!</p>
          <NavLink to="/user-form">
            <button className="about-button">Client</button>
          </NavLink>
        </section>
      </section>
      <p className="second">Then, Sign-In to your account </p>
      <p> or Sign-Up if you haven't had the opportunity yet!</p>
      <p className="third">
        Explore your profile and edit your settings!
      </p>
    </section>
  );
};

export default About;
