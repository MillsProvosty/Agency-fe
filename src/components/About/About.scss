.about {
    font-size: 1em;

    & span {
        font-weight:bold
    }

    & .first {
        margin-top: 1em;
        font-size: 2.5em;
    }

    & .second, .third {
        margin-top: 2em;
    }

    & .third {
        margin-bottom: 2em;
    }

    & .options {
        margin-top: 2em;
        display: flex;

        & .option-1, .option-2 {
            margin-top: 1em;
            width: 50%
        }

        & button {
            margin-top: 1em;
        }

    }

    & .option-1 {
        border-right: 1px solid darkblue
    }
}

@media screen and (max-width: 375px) {
    html, body {
        overflow-x: hidden;
      }
      body {
        position: relative;
      }
    .about {
        & .options {
            margin-top: 1rem;
            display:block;
            justify-content: center;
            & .option-1, .option-2 {
                margin: auto;
                width: 80%
            }
            & .option-1 {
                border-bottom: 1px solid darkblue;
                border-right: none; 
            }
        }
    }
}