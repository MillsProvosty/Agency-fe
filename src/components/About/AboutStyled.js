import styled from "styled-components";

export const AboutSection = styled.section`
  font-size: 1em;
  background-color: white;
  @media screen and (max-width: 375px) {
    font-size: .75em;
  }
`;

export const Bold = styled.span`
  font-weight: bold;
`;

export const BigLetters = styled.p`
  margin-top: ${props => (props.first ? "1em" : "2em")};
  font-size: ${props => (props.first ? "2.5em" : "1.5em")};
  margin-bottom: ${props => (props.third ? "2em" : "0em")};

  @media screen and (max-width: 375px) {
    margin-top: ${props => (props.first ? "1em" : "2em")};
    font-size: ${props => (props.first || props.second ? "1.8em" : ".75em")};
    margin-bottom: ${props => (props.third? "2em": "0em")};
  }
`;

export const Options = styled.section`
  margin-top: 2em;
  display: flex;
  @media screen and (max-width: 375px) {
    margin-top: 1rem;
    display:block;
    justify-content: center;
  }
  @media screen and (display-mode: standalone) {
    margin-top: 1rem;
    display:block;
    justify-content: center;
  }
`;

export const MoreOptions = styled.section`
  margin-top: 1em;
  width: 50%;
  border-right: ${props => (props.one ? "1px solid #7A86CB" : "none")};
  @media screen and (max-width: 375px) {
    margin: auto;
    width: 80%
    border-bottom: ${props => (props.one ? "1px solid #7A86CB": "none")}
    border-right: ${props => (props.one ? "none": '')}
  }
  @media screen and (display-mode: standalone) {
    margin: auto;
    width: 80%
    border-bottom: ${props => (props.one ? "1px solid #7A86CB": "none")}
    border-right: ${props => (props.one ? "none": '')}
  }
`;

export const PTag = styled.p`
  font-size: 1.5em;
`;

export const QuickSection = styled.section`
  display: flex;
  width: 75%;
  margin: auto
`


export const Button = styled.button`
  color: white;
  background-color: #7A86CB;
  border-radius: 5px;
  font-size: 2em;
  width: 175px;
  font-family: "Quicksand", sans-serif;
  border: 2px solid white;
  margin-top: 30px;
  margin-bottom: 20px;
  margin-right: ${props => (props.volunteer ? "20px" : "0px")};
  margin-left: ${props => (props.client ? "20px" : "0px")}

  :hover {
    border: 2px solid #7A86CB;
    color: #7A86CB;
    background-color: white;
}
`;
