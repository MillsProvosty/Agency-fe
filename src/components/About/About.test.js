import React from 'react';
import About from './About';
import { shallow } from 'enzyme';

describe('About', () => {

  it('should match snapshot', () => {
    const wrapper = shallow(
      <About />
    )
    expect(wrapper).toMatchSnapshot()
  })
})