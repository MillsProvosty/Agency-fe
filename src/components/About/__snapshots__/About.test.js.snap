// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About should match snapshot 1`] = `
<styled.section>
  <Nav />
  <styled.p
    first={true}
  >
    <styled.span>
      Agency
    </styled.span>
     is a support networking service.
  </styled.p>
  <styled.p
    second={true}
  >
    To begin..
  </styled.p>
  <styled.section>
    <styled.section
      one={true}
    >
      <styled.p>
        select the 
        <styled.span>
          Volunteer
        </styled.span>
         button
      </styled.p>
      <styled.p>
        if you can lend a hand...
      </styled.p>
      <NavLink
        to="/user-form"
      >
        <styled.button>
          Volunteer
        </styled.button>
      </NavLink>
    </styled.section>
    <styled.section>
      <styled.p>
        or, select the 
        <styled.span>
          Client
        </styled.span>
         button
      </styled.p>
      <styled.p>
        if you would like some help!
      </styled.p>
      <NavLink
        to="/user-form"
      >
        <styled.button>
          Client
        </styled.button>
      </NavLink>
    </styled.section>
  </styled.section>
  <styled.p
    second={true}
  >
    Then, Sign-In to your account
  </styled.p>
  <styled.p>
     or Sign-Up if you haven't had the opportunity yet!
  </styled.p>
  <styled.p
    second={true}
    third={true}
  >
    Explore your profile and edit your settings!
  </styled.p>
</styled.section>
`;
